Tasks/DownloadPipelineArtifactV2/task.json

```diff
@@ -0,0 +1,229 @@
+            }
+        },
+        {
+            "name": "runBranch",
+            "aliases": [
+                "branchName"
+            ],
+            "type": "string",
+            "label": "Branch name",
+            "defaultValue": "refs/heads/master",
+            "visibleRule": "source == specific && runVersion == latestFromBranch",
+            "required": true,
+            "helpMarkDown": "Specify to filter on branch/ref name, for example: ```refs/heads/develop```."
+        },
+        {
+            "name": "runId",
+            "aliases": [
+                "pipelineId",
+                "buildId"
+            ],
+            "type": "pickList",
+            "label": "Build",
+            "defaultValue": "",
+            "required": true,
+            "visibleRule": "source == specific && runVersion == specific",
+            "properties": {
+                "EditableOptions": "True",
+                "DisableManageLink": "True"
+            },
+            "helpMarkDown": "The build from which to download the artifacts."
+        },
+        {
+            "name": "tags",
+            "type": "string",
+            "label": "Build Tags",
+            "defaultValue": "",
+            "visibleRule": "source == specific && runVersion != specific",
+            "required": false,
+            "helpMarkDown": "A comma-delimited list of tags. Only builds with these tags will be returned."
+        },
+        {
+            "name": "artifact",
+            "aliases": [
+                "artifactName"
+            ],
+            "type": "pickList",
+            "label": "Artifact name",
+            "defaultValue": "",
+            "required": false,
+            "properties": {
+                "EditableOptions": "True",
+                "DisableManageLink": "True"
+            },
+            "helpMarkDown": "The name of the artifact to download."
+        },
+        {
+            "name": "patterns",
+            "aliases": [
+                "itemPattern"
+            ],
+            "type": "multiLine",
+            "label": "Matching patterns",
+            "defaultValue": "**",
+            "required": false,
+            "properties": {
+                "rows": "3",
+                "resizable": "true"
+            },
+            "helpMarkDown": "One or more file matching patterns (new line delimited) that limit which files get downloaded."
+        },
+        {
+            "name": "path",
+            "aliases": [
+                "targetPath",
+                "downloadPath"
+            ],
+            "type": "string",
+            "label": "Destination directory",
+            "defaultValue": "$(Pipeline.Workspace)",
+            "required": true,
+            "helpMarkDown": "Directory to download the artifact files to. Can be relative to the pipeline workspace directory or absolute. If more than one artifact is selected for download, a sub-directory will be created for each."
```

Comment:
>If more than one artifact is selected for download [](start = 139, length = 50)

There's no way to select multiple artifacts. The only way is to have the artifact field empty.

Reply:
Updated it to "Directory to download the artifact files to. Can be relative to the pipeline workspace directory or absolute. If multi-download option is applied (by leaving an empty artifact name), a sub-directory will be created for each."